### ISSUE TITLE
Data points plotted incorrect order

### ISSUE THREAD

**Issue by ln-komandur** [Rello/analytics#62]
I am plotting 2 data points on a graph from a refrigerator door monitor. 

1. One is temperature in celcius. It is plotted as a line with data coming in at regular intervals. 
2. The second one is the position of the fridge door. It is plotted as a bar with +5 for an open door, and -5 for a closed door. This is plotted only when that event happens. i.e. irregular intervals, but also has the timestamp.

Both data have timestamp on the X axis.

I find that the second data (+5 or -5 on the bar) is plotted without synchronizing its timestamp even though the data table shows the values correctly. Please see the screenshots attached here. Even if I change the bar to line, its the same behaviour.

![Graph on Nextcloud Analytics app](https://user-images.githubusercontent.com/68081200/90636290-00cc2180-e1f0-11ea-8054-3e9005b5cbc7.png)

The data below is sorted by time

![Data Ordered by Timestamp](https://user-images.githubusercontent.com/68081200/90636766-b5664300-e1f0-11ea-8e8e-41124f2c3a80.png)

**Comment by Rello** [comment]
Hello,
what charttype did you use? did you use it to timeline?

**Comment by ln-komandur** [comment]
Thanks for your response. I had not. And then I made the change after your question.
However, I am only getting an empty graph though there is data in the table below.

![timeline](https://user-images.githubusercontent.com/68081200/90642642-442a8e00-e1f8-11ea-976f-fedbb31e4847.png)


Irrespective of the charttype, should the data not be interleaved by default like in the table?
In my example, it is timeline. There could be another in which one data has even numbers for X-axis and another has odd numbers for X-axis. And one of them could be irregular also.

**Comment by Rello** [comment]
hey,
can you copy past some of your records here? then I don´t need to type them.
copy/paste from the table.

one hint: did you try to set the celsius series to the secondary y-axis already? should fit pretty well

**Comment by ln-komandur** [comment]
Thanks for your response. I tried Celsius on secondary y-axis already. Only the y-axis changes, there is no data that gets plotted whether I choose charttype as line or timeline. Please see screenshot here

![Celsius on secondary](https://user-images.githubusercontent.com/68081200/90685805-f16cc880-e22f-11ea-8f0e-a67db60def8f.png)



My data is as below. I guess it should look good if you copy paste to notepad

Event	Time	Value
Celcius	07:21:12	22
Door	07:21:12	-5
Celcius	07:21:24	20
Celcius	07:21:43	19
Celcius	07:21:57	18
Celcius	07:22:13	17
Celcius	07:22:36	16
Celcius	07:22:55	15
Celcius	07:23:23	14
Celcius	07:23:58	13
Celcius	07:24:47	12
Celcius	07:25:49	12
Door	07:25:49	5
Celcius	07:26:02	12
Door	07:26:02	-5
Celcius	07:26:14	11
Celcius	07:27:17	10
Celcius	07:28:29	10
Door	07:28:29	5
Celcius	07:28:41	13
Door	07:28:41	-5

**Comment by Rello** [comment]
ok, here you go. let me explain:
for a non-time-chart, the x-axis values have to be the same. because a chart is generated by an array for the x-axis-values (e.g. 1,2,3,4,5) and then the corresponsing arrays for the datasets for the y-axis. if they are not the same count, it will fill them from the left (100,200,300,400,500) and (10,30,50). they will never match unless the second dataset is filled with blank values.

tl;dr: this type does not work here.

you need the timeline, because here every point is painted individually by an x/y coordinate.
In your case, the time format is no standard date value and the backend does not know how to read it. while it is difficult to track this automatically, there is a way to customize the chart generation.

In the report settings, use the chart options field:
![Bildschirmfoto 2020-08-25 um 10 04 07](https://user-images.githubusercontent.com/13385119/91148712-63269580-e6ba-11ea-9dfa-e61eaa11b5bf.png)

by typing in the following, you tell the chart your format:
`{"scales":{"xAxes":[{"time":{"parser":"hh:mm:ss"}}]}}`

please remove the existing values in the both options fields before you enter it. afterwards you can reconfigure the chart and it will look like that:
![Bildschirmfoto 2020-08-25 um 10 07 34](https://user-images.githubusercontent.com/13385119/91148996-cdd7d100-e6ba-11ea-886c-97732c9a93d7.png)

let me know.
Unfortunately its difficult to detect you special case out of the box. I hope this advanced options help?
let me know

**Comment by ln-komandur** [comment]
Thanks very much @Rello . Your clarification worked perfectly.

Earlier when I had the chart type set as "line", (because no data was getting plotted when I put it as "timeline") I got the below.
![Before - with Chart type as Line](https://user-images.githubusercontent.com/68081200/91761462-8e1d5780-eb9a-11ea-9604-08a6d0d81dc7.png)


Now, when I changed the chart type to "timeline" and put the chart format as you suggested in the "Chart options", and put Door data series as "Secondary" in the Vertical axis, I got the below. This is exactly what I wanted. Thanks again.

![After - Charttype is Timeline - Chart options - secondary axis](https://user-images.githubusercontent.com/68081200/91763853-d721db80-eb9b-11ea-8df3-e32ffb67b45c.png)


In the meantime, I had made a short video to demo the application I was building. You may be interested to see it here https://youtu.be/InoAnvhIq6A    !

I will get back to you on other items you are currently testing (data deletion, posting arrays, using names instead of dimension1 etc.). Thanks
